File: src/main/java/com/qtech/bubbles/core/handler/Handler.java
package com.qtech.bubbles.core.handler;

import com.qtech.bubbles.entity.Entity;
import com.qtech.bubbles.entity.player.PlayerEntity;

import java.util.LinkedList;

/**
 * @author Quinten Jungblut
 */
public class Handler {
    final LinkedList<Entity> object = new LinkedList<>();

    /**
     * @author Quinten Jungblut
     */
    @SuppressWarnings("EmptyMethod")
    public void tick() {
//        for (int i = 0; i < object.size(); i++) {
//            GameObject tempObject = object.get(i);
//            tempObject.tick();
//        }
    }

    /**
     * @author Quinten Jungblut
     */
    @SuppressWarnings("EmptyMethod")
    public void render() {
//        for (int i = 0; i  < object.size(); i++) {
//            GameObject tempObject = object.get(i);
//
//            tempObject.render(g);
//        }
    }

    /**
     * @author Quinten Jungblut
     */
    public void addObject(Entity object) {
        this.object.add(object);
    }

    public void removeObject(Entity object) {
        this.object.remove(object);
    }

    public void clearEnemies() {
        for (int i = 0; i < object.size(); i++) {
            Entity tempObject = object.get(i);

            if (tempObject instanceof PlayerEntity) {
                object.clear();
                addObject(tempObject);
            }
        }
    }
}

File: src/main/kotlin/qtech/bubbles/core/handler/Handler.kt
package qtech.bubbles.core.handler

import qtech.bubbles.entity.Entity
import qtech.bubbles.entity.player.PlayerEntity
import java.util.*

/**
 * @author Quinten Jungblut
 */
class Handler {
    val `object` = LinkedList<Entity>()

    /**
     * @author Quinten Jungblut
     */
    fun tick() {
//        for (int i = 0; i < object.size(); i++) {
//            GameObject tempObject = object.get(i);
//            tempObject.tick();
//        }
    }

    /**
     * @author Quinten Jungblut
     */
    fun render() {
//        for (int i = 0; i  < object.size(); i++) {
//            GameObject tempObject = object.get(i);
//
//            tempObject.render(g);
//        }
    }

    /**
     * @author Quinten Jungblut
     */
    fun addObject(`object`: Entity) {
        this.`object`.add(`object`)
    }

    fun removeObject(`object`: Entity) {
        this.`object`.remove(`object`)
    }

    fun clearEnemies() {
        for (i in `object`.indices) {
            val tempObject = `object`[i]
            if (tempObject is PlayerEntity) {
                `object`.clear()
                addObject(tempObject)
            }
        }
    }
}